editor/components/block-switcher/index.js

```diff
@@ -102,21 +96,19 @@ export function BlockSwitcher( { blocks, onTransform, isLocked } ) {
 }
 
 export default compose(
-	connect(
-		( state, ownProps ) => {
-			return {
-				blocks: ownProps.uids.map( ( uid ) => getBlock( state, uid ) ),
-			};
+	withSelect( ( select, ownProps ) => {
+		return {
+			blocks: ownProps.uids.map( ( uid ) => select( 'core/editor' ).getBlock( uid ) ),
```

Comment:
Not related to your change, but `map` usage in here looks suspicious. It probably rerenders too often.

Reply:
Now I disagree with myself, as long as it is the only prop passed, it is fine :)